### PR TITLE
Add VMC Oauth app support

### LINKED ISSUE #944: Oauth App Support for VMConAWS authentication
### Is your feature request related to a problem? Please describe.

NSXT Terraform already supports using the CSP refresh token to obtain an access token from CSP, which is then used for authenticating to VMC NSX. Similarly, OAuth2 is another method of obtaining an access token.

https://blogs.vmware.com/cloud/2023/03/28/announcing-oauth-app-support-for-vmware-cloud-automation-tools/

This issue is to add this additional method

### Describe the solution you'd like

Implementation of OAuth App Support for the NSX-T Terraform Provider for VMConAWS environmnents:
https://blogs.vmware.com/cloud/2023/03/28/announcing-oauth-app-support-for-vmware-cloud-automation-tools/

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by wsquan171**
This change adds support of NSX token exchange on VMC using Oauth app, which is a newer method compared to using refresh token.

Implementation mainly referenced [python client for vmc on aws](https://github.com/vmware/python-client-for-vmware-cloud-on-aws/blob/development/pyvmc_fxns.py#L166) and terraform vmc provider.

Testing done:
With Oauth app created in server-to-server mode, assigned with all roles under "VMware Cloud on AWS" category, provider configured as:

```
provider "nsxt" {
  host                 = "<nsx-private-ip>.rp.vmwarevmc.com/vmc/reverse-proxy/api/orgs/<org-id>/sddcs/<ssdc-id>/sks-nsxt-manager"
  allow_unverified_ssl = true
  max_retries          = 2
 
  enforcement_point    = "vmc-enforcementpoint"
  vmc_client_id        = "<oauth-app-id>"
  vmc_client_secret    = "<oauth-app-secret>"
}
``` 

Able to create and delete resources like:
```
resource "nsxt_policy_gateway_qos_profile" "test" {
  display_name        = "test"
  description         = "Terraform provisioned profile"
  burst_size          = 10
  committed_bandwidth = 20
}
```

For regression, verified using access token to create and delete the same resource. The provider is configured as follows:
```
provider "nsxt" {
  host                 = "<nsx-private-ip>.rp.vmwarevmc.com/vmc/reverse-proxy/api/orgs/<org-id>/sddcs/<ssdc-id>/sks-nsxt-manager"
  allow_unverified_ssl = true
  max_retries          = 2
 
  enforcement_point    = "vmc-enforcementpoint"
  vmc_token            = "<access-token>"
}
```

e2e test covers non-vmc cases.

Closes #944

### CODE REVIEW

File: nsxt/provider.go [hunk 1]
```diff
@@ -158,13 +158,30 @@ func Provider() *schema.Provider {
 				Type:        schema.TypeString,
 				Optional:    true,
 				Description: "URL for VMC authorization service (CSP)",
-				DefaultFunc: schema.EnvDefaultFunc("NSXT_VMC_AUTH_HOST", "console.cloud.vmware.com/csp/gateway/am/api/auth/api-tokens/authorize"),
```
Comment (annakhm) [inline nsxt/provider.go:161]: I'm afraid we're not allowed to change defaults in order not to break backward compatibility. Is this change required?

Comment (wsquan171) [reply nsxt/provider.go:161]: Technically the backward compatibility is not broken in that the same config (leaving out auth host in both config and env) will still work. The same default value is set in another place.

The reason I need to change this, is because oauth and access token need to use different CSP API URL to exchange for api token. The default value now needs to be adjusted based on auth method used (access token or oauth app). I checked that currently from schema this field is consumed in `getAPIToken`. I think it should be safe.

**Review by annakhm — APPROVED** [review]

File: website/docs/index.html.markdown [hunk 2]
```diff
@@ -221,7 +221,15 @@ The following arguments are used to configure the VMware NSX-T Provider:
   partially successful realization as valid state and not fail apply.
 * `vmc_token` - (Optional) Long-lived API token for authenticating with VMware
   Cloud Services APIs. This token will be used to short-lived token that is
-  needed to communicate with NSX Manager in VMC environment.
+  needed to communicate with NSX Manager in VMC environment. In conflict with
```
Comment (annakhm) [inline website/docs/index.html.markdown:224]: Just cosmetic comment - I'm not sure `in conflict` is a term the user will always understand. Perhaps `can not be specified together with XXX`?
